src/controller/c/supervisor.c

```diff
@@ -1077,25 +1103,44 @@ static void supervisor_read_answer(WbDevice *d, WbRequest *r) {
       for (i = 0; i < 3; i++)
         center_of_mass_node_ref->center_of_mass[i] = request_read_double(r);
       break;
-    case C_SUPERVISOR_NODE_GET_CONTACT_POINTS:
-      free(contact_points_node_ref->contact_points);
-      free(contact_points_node_ref->node_id_per_contact_points);
-      contact_points_node_ref->contact_points = NULL;
-      contact_points_node_ref->node_id_per_contact_points = NULL;
-      contact_points_node_ref->number_of_contact_points = request_read_int32(r);
-      if (contact_points_node_ref->number_of_contact_points > 0) {
-        const int three_times_size = 3 * contact_points_node_ref->number_of_contact_points;
-        contact_points_node_ref->contact_points = malloc(three_times_size * sizeof(double));
-        contact_points_node_ref->node_id_per_contact_points =
-          malloc(contact_points_node_ref->number_of_contact_points * sizeof(int));
-        for (i = 0; i < contact_points_node_ref->number_of_contact_points; i++) {
-          contact_points_node_ref->contact_points[3 * i] = request_read_double(r);
-          contact_points_node_ref->contact_points[3 * i + 1] = request_read_double(r);
-          contact_points_node_ref->contact_points[3 * i + 2] = request_read_double(r);
-          contact_points_node_ref->node_id_per_contact_points[i] = request_read_int32(r);
+    case C_SUPERVISOR_NODE_GET_CONTACT_POINTS: {
+      const int node_id = request_read_int32(r);
+      const int include_descendants = (int)request_read_uchar(r);
+
+      WbNodeRef contact_point_node = NULL;
+      if (contact_points_node_ref && contact_points_node_ref->id == node_id &&
+          contact_points_node_ref->contact_points_include_descendants == include_descendants)
+        contact_point_node = contact_points_node_ref;
+      else {
+        WbNodeRef tmp_node = node_list;
+        while (tmp_node) {
+          if (tmp_node->id == node_id) {
+            contact_point_node = tmp_node;
+            break;
+          }
+          tmp_node = tmp_node->next;
         }
       }
+      assert(contact_point_node);
```

Comment:
@omichel Is it possible it is crashing here?

Reply:
It's actually not crashing any more...